sources/packages/api/src/route-controllers/student/student.students.controller.ts

```diff
@@ -21,28 +53,150 @@ import { ClientTypeBaseRoute } from "../../types";
 @ApiTags(`${ClientTypeBaseRoute.Student}-students`)
 @Injectable()
 export class StudentStudentsController extends BaseController {
-  constructor(private readonly fileService: StudentFileService) {
+  constructor(
+    private readonly fileService: StudentFileService,
+    private readonly studentControllerService: StudentControllerService,
+    private readonly gcNotifyActionsService: GCNotifyActionsService,
+    private readonly applicationService: ApplicationService,
+    private readonly studentService: StudentService,
+  ) {
     super();
   }
 
   /**
-   * This controller returns all student documents uploaded
-   * by student uploader.
+   * Gets all student documents uploaded to the student account.
    * @returns list of student documents.
    */
   @Get("documents")
-  @ApiNotFoundResponse({
-    description: "The user does not have a student account associated with.",
-  })
   async getStudentFiles(
     @UserToken() userToken: StudentUserToken,
-  ): Promise<StudentUploadFileDTO[]> {
+  ): Promise<StudentUploadFileAPIOutDTO[]> {
     const studentDocuments = await this.fileService.getStudentUploadedFiles(
       userToken.studentId,
     );
     return studentDocuments.map((studentDocument) => ({
       fileName: studentDocument.fileName,
       uniqueFileName: studentDocument.uniqueFileName,
+      fileOrigin: studentDocument.fileOrigin,
     }));
   }
+
+  /**
+   * Gets a student file and writes it to the HTTP response.
+   * @param uniqueFileName unique file name (name+guid).
+   * @param response file content.
+   */
+  @Get("files/:uniqueFileName")
+  @ApiNotFoundResponse({
+    description:
+      "Requested file was not found or the user does not have access to it.",
+  })
+  async getUploadedFile(
+    @UserToken() userToken: StudentUserToken,
+    @Param("uniqueFileName") uniqueFileName: string,
+    @Res() response: Response,
+  ): Promise<void> {
+    await this.studentControllerService.writeFileToResponse(
+      response,
+      uniqueFileName,
+      userToken.studentId,
+    );
+  }
+
+  /**
+   * Allow files uploads to a particular student.
+   * @param userToken authentication token.
```

Comment:
User token is added to the comment in this method. But not in previous one. We can follow one or the other way.